metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/EntityResource.java

```diff
@@ -389,36 +417,94 @@ public Task<RollbackResponse> deleteEntities(@ActionParam("registryId") @Optiona
     return RestliUtil.toTask(() -> {
+      // Find the timeseries aspects to delete. If aspectName is null, delete all.
+      List<String> timeseriesAspectNames =
+          EntitySpecUtils.getEntityTimeseriesAspectNames(_entityService.getEntityRegistry(), urn.getEntityType());
+      if (aspectName != null && !timeseriesAspectNames.contains(aspectName)) {
+        throw new IllegalArgumentException(String.format("Failed to find timeseries aspect {}", aspectName));
+      }
+      List<String> timeseriesAspectsToDelete =
+          (aspectName == null) ? timeseriesAspectNames : ImmutableList.of(aspectName);
+
       DeleteEntityResponse response = new DeleteEntityResponse();
-      RollbackRunResult result = _entityService.deleteUrn(urn);
+      if (aspectName == null) {
+        RollbackRunResult result = _entityService.deleteUrn(urn);
+        response.setRows(result.getRowsDeletedFromEntityDeletion());
+      }
+      Long numTimeseriesDocsDeleted =
+          deleteTimeseriesAspects(urn, startTimeMills, endTimeMillis, timeseriesAspectsToDelete);
+      log.info("Total number of timeseries aspect docs deleted: {}", numTimeseriesDocsDeleted);
 
       response.setUrn(urnStr);
-      response.setRows(result.getRowsDeletedFromEntityDeletion());
+      response.setTimeseriesRows(numTimeseriesDocsDeleted);
 
       return response;
     }, MetricRegistry.name(this.getClass(), "delete"));
   }
 
+  private Long deleteTimeseriesAspects(@Nonnull Urn urn, @Nullable Long startTimeMillis, @Nullable Long endTimeMillis,
+      @Nonnull List<String> aspectsToDelete) {
+    long totalNumberOfDocsDeleted = 0;
+    // Construct the filter.
+    List<Criterion> criteria = new ArrayList<>();
+    final Criterion hasUrnCriterion =
+        new Criterion().setField("urn").setCondition(Condition.EQUAL).setValue(urn.toString());
+    criteria.add(hasUrnCriterion);
+
+    if (startTimeMillis != null) {
+      final Criterion startTimeCriterion = new Criterion().setField(ES_FILED_TIMESTAMP)
+          .setCondition(Condition.GREATER_THAN_OR_EQUAL_TO)
+          .setValue(startTimeMillis.toString());
+      criteria.add(startTimeCriterion);
+    }
+    if (endTimeMillis != null) {
+      final Criterion endTimeCriterion = new Criterion().setField(ES_FILED_TIMESTAMP)
+          .setCondition(Condition.LESS_THAN_OR_EQUAL_TO)
+          .setValue(endTimeMillis.toString());
+      criteria.add(endTimeCriterion);
+    }
+    final Filter filter = QueryUtils.getFilterFromCriteria(criteria);
+
+    // Delete all the timeseries aspects by the filter.
+    final String entityType = urn.getEntityType();
+    for (final String aspect : aspectsToDelete) {
+      DeleteAspectValuesResult result = _timeseriesAspectService.deleteAspectValues(entityType, aspect, filter);
+      totalNumberOfDocsDeleted += result.getNumDocsDeleted();
+
+      log.info("Number of timeseries docs deleted for entity:{}, aspect:{}, urn:{}, startTime:{}, endTime:{}={}",
```

Comment:
You are info logging a lot. Remember there was one inside the delet aspect values. Can we make this log.debug?

Reply:
Fixed.